tests/validation/tests/v3_api/test_certificate.py

```diff
@@ -0,0 +1,571 @@
+    else:
+        return get_ssh_key(key_name)
+
+
+rancher_private_key = get_private_key('RANCHER_VALID_CERT_PRIV_KEY',
+                                      'privkey.pem')
+rancher_cert = get_private_key('RANCHER_VALID_CERT', 'fullchain.pem')
+rancher_ssc_private_key = get_private_key('RANCHER_SSC_CERT_PRIV_KEY',
+                                          'key.pem')
+rancher_ssc_cert = get_private_key('RANCHER_SSC_CERT', 'cert.pem')
+
+rbac_role_list = [
+                  CLUSTER_OWNER,
+                  CLUSTER_MEMBER,
+                  PROJECT_OWNER,
+                  PROJECT_MEMBER,
+                  PROJECT_READ_ONLY
+                 ]
+
+
+@pytest.mark.usefixtures("create_project_client")
+class TestCertificate:
+
+    @pytest.fixture(autouse="True")
+    def certificate_test_setup(self):
+        """
+        Test set up which runs before and after all the tests in the class
+        Creates Workload_2 if required and delete all the workload and ingres
+        created after test execution.
+        """
+        self.p_client = namespace["p_client"]
+        self.ns = namespace["ns"]
+        self.c_client = namespace["c_client"]
+        self.cluster = namespace["cluster"]
+        self.project = namespace["project"]
+        self.certificate_valid = namespace["cert_valid"]
+        self.certificate_ssc = namespace["cert_ssc"]
+        self.certificate_all_ns_valid = namespace["cert_allns_valid"]
+        self.certificate_all_ns_ssc = namespace["cert_allns_ssc"]
+        self.node_id = namespace["node_id"]
+        wl_name = random_test_name("workload-test")
+        wl_con = [{"name": "wk1-test",
+                   "image": TEST_IMAGE}]
+        scheduling = {"node": {"nodeId": self.node_id}}
+        self.workload = self.p_client.create_workload(
+            name=wl_name, containers=wl_con, namespaceId=self.ns.id,
+            scheduling=scheduling
+        )
+        self.ingress = None
+        self.workload_2 = None
+        yield
+        self.p_client.delete(self.workload)
+        if self.workload_2 is not None:
+            self.p_client.delete(self.workload_2)
+        if self.ingress is not None:
+            self.p_client.delete(self.ingress)
+
+    def test_certificate_create_validcert_for_single_ns(self):
+        """
+        Test steps:
+        1. Validate the workload available in ns-certificate namespace
+        2. Create an ingress including trusted certificate scoped for current
+        namespace and route53 host.
+        3. validate the ingress using endpoint
+        """
+        ingress_name = random_test_name("ingress-test")
+        host = route_entry_53_1
+        path = "/name.html"
+        rule = {"host": host,
+                "paths": [{"path": path, "workloadIds": [self.workload.id],
+                           "targetPort": "80"}]}
+        tls = {"certificateId": self.certificate_valid.id, "hosts": [host]}
+        validate_workload(self.p_client, self.workload, "deployment",
```

Comment:
what is the purpose of validating the workload here?

Reply:
Workload is created in test setup for each test, so just making sure it is working before creating an ingress pointing to it.